app/services/merge_patient_service.rb

```diff
@@ -74,7 +71,42 @@ def merge_business_identifiers(business_identifier_params, patient)
     end
   end
 
-  def log_update_discarded_patient(merged_patient)
-    NewRelic::Agent.increment_metric("MergePatientService/update_discarded_patient") if merged_patient.discarded?
+  def discard_patient_data(patient)
+    if patient.deleted_at.present? && existing_patient&.deleted_at.nil?
+      # Patient has been soft-deleted by the client, server should soft-delete the patient and their associated data.
+      patient.update(deleted_by_user_id: request_metadata[:request_user_id])
+      patient.discard_data
+    end
+  end
+
+  def touch_patient(patient, address, phone_numbers, business_ids)
+    if (address.present? && address.merged?) || phone_numbers.any?(&:merged?) || business_ids.any?(&:merged?)
+      patient.touch
+      #
+      # This is a rare scenario that might be possible in the future.
+      # If the client allows the user to update the patient's address or phone_number,
+      # there might be a case where the address or phone_number for a patient is updated for a discarded patient.
+      #
+      # These updates should not ideally be made at all because they will be invisible to the user.
+      #
+      # We can fix this issue, but it requires re-working the merge function, so we'll currently just
+      # track the incidence rate, so we can plan for a fix if necessary.
+      log_update_discarded_patient if patient.discarded?
+    end
+  end
+
+  def new_patient_metadata
+    {
+      registration_facility_id: payload[:registration_facility_id].presence || request_metadata[:request_facility_id],
```

Comment:
Here, now, it's explicit what the `request_facility_id` (headers) is and what `registration_facility_id` is.